### DISCUSSION TITLE
Unable to import variables and functions from firedrake module

### DISCUSSION THREAD

**Discussion by huisophiewang** [firedrakeproject/firedrake#3927] [Firedrake support] [answered]
Hi,

I installed the latest version of firedrake with Python 3.12. When I run "from firedrake import Mesh", it has "ImportError: cannot import name 'Mesh' from 'firedrake' (unknown location)". When I use "dir(firedrake)" to list the names, it only has these names ['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']. Any idea about what the problem might be and how to solve it?

**Comment by connorjward** [comment]
I think this may suggest that the installation failed. Can you share your log files?

**Reply by huisophiewang** [reply]
Here's the log. [firedrake-install.log](https://github.com/user-attachments/files/18118514/firedrake-install.log)

**Reply by huisophiewang** [reply]
I have the same import error when using the docker image.

**Reply by connorjward** [reply]
Just to check, have you activated the virtual environment?

**Reply by huisophiewang** [reply]
I believe so. Here's the screenshot
![image](https://github.com/user-attachments/assets/b093cf55-9e5e-4ede-88dd-7877d287cbe7)

**Reply by connorjward** [reply, marked as answer]
Can you try running this command from a different directory? (e.g. `cd ..; python -c "from firedrake import *"`). Just to make sure that the top-level `firedrake` venv isn't being recognised as a package.

**Reply by huisophiewang** [reply]
That works! Thank you so much!
